File: src/backend/src/Hotel5000_Api/Core/Entities/Domain/Lodging.cs
using System.Collections.Generic;

namespace Core.Entities.Domain
{
    public class Lodging : BaseEntity
    {
        public string Name { get; set; }
        public int UserId { get; set; }
        public int LodgingTypeId { get; set; }

        public User User { get; set; }
        public LodgingType LodgingType { get; set; }
        public ICollection<LodgingAddress> LodgingAddresses { get; set; }
        public ICollection<Room> Rooms { get; set; }
        public ICollection<ReservationWindow> ReservationWindows { get; set; }
    }
}

File: src/frontend/hotel5000angular/src/app/models/Lodging.ts
import { LodgingAddress } from './LodgingAddress';
import { ReservationWindow } from './ReservationWindow';
import { Room } from './Room';

export class Lodging{
    id: number;
    name: string;
    lodgingType: string;
    userId: number;
    addresses: LodgingAddress[];
    rooms: Room[];
    reservationWindows: ReservationWindow[];
}